zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,165 @@
+ * // Session Timeout
+ * .sessionTimeOut(10000)
+ * // NodeValueCodec
+ * .nodeValueCodec(NodeValueCodec.DEFAULT)
+ * .build();
+ * // Set to `Server`.
+ * Server server = ...
+ * server.serverListener(register);
+ * }</pre>
+ *
+ * <p>You can also specify the {@link CuratorFramework} instance to use. In this case,
+ * {@link #connectTimeout(int)} and {@link #sessionTimeout(int)} will be ignored.
+ *
+ * <h2>Example</h2>
+ * <pre>{@code
+ * // CuratorFramework instance and Zookeeper Node to use
+ * ServerListener register = new ZooKeeperUpdatingListenerBuilder(curatorFramework, "/myProductionEndpoints")
+ * // NodeValueCodec
+ * .nodeValueCodec(NodeValueCodec.DEFAULT)
+ * .build();
+ * // Set to `Server`.
+ * Server server = ...
+ * server.serverListener(register);
+ * }</pre>
+ * */
+public final class ZooKeeperUpdatingListenerBuilder {
+    private CuratorFramework client;
+    private String connect;
+    private String node;
+    private int connectTimeout = Constants.DEFAULT_CONNECT_TIMEOUT;
+    private int sessionTimeout = Constants.DEFAULT_SESSION_TIMEOUT;
+    private Endpoint endpoint;
+    private NodeValueCodec nodeValueCodec = NodeValueCodec.DEFAULT;
+
+    /**
+     * Creates a {@link ZooKeeperUpdatingListenerBuilder} with a {@link CuratorFramework} instance and a zNode.
+     *
+     * @param client the curator framework instance.
+     */
+    public ZooKeeperUpdatingListenerBuilder(CuratorFramework client, String node) {
+        this.client = requireNonNull(client, "client");
+        checkArgument(!isNullOrEmpty(node), "node");
+        this.node = node;
+    }
+
+    /**
+     * Creates a {@link ZooKeeperUpdatingListenerBuilder} with a Zookeeper Connection String and a zNode.
+     *
+     * @param connect the zookeeper connection string.
+     * @param node the zookeeper node to register.
+     */
+    public ZooKeeperUpdatingListenerBuilder(String connect, String node) {
+        checkArgument(!isNullOrEmpty(connect), "connect");
+        checkArgument(!isNullOrEmpty(node), "node");
+        this.connect = connect;
+        this.node = node;
+    }
+
+    /**
+     * Sets the connect timeout (in ms). (default: 1000)
+     *
+     * <p>Ignored when {@link CuratorFramework} is set. (see: {@link #client})
+     *
+     * @param timeout the connect timeout.
+     */
+    public ZooKeeperUpdatingListenerBuilder connectTimeout(int timeout) {
+        if (client != null) {
+            throw new IllegalStateException("client is already set");
+        }
+        checkArgument(timeout > 0, "timeout: %s (expected: > 0)", timeout);
+        this.connectTimeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the session timeout (in ms). (default: 10000)
+     *
+     * <p>Ignored when {@link CuratorFramework} is set. (see: {@link #client})
+     *
+     * @param timeout the session timeout.
+     */
+    public ZooKeeperUpdatingListenerBuilder sessionTimeout(int timeout) {
```

Comment:
- sessionTimeout -> sessionTimeoutMillis
- Please add `sessionTimeout(Duration timeout)` as well.
- Please change the type of `timeout` to `long`, and convert it to `Integer.MAX_VALUE` if it's beyond the limit.